packages/pigeon/platform_tests/test_plugin/linux/pigeon/core_tests.gen.h

```diff
@@ -0,0 +1,1998 @@
+    CORE_TESTS_PIGEON_TEST,
+    HOST_INTEGRATION_CORE_API_ECHO_NULLABLE_MAP_RESPONSE, GObject)
+
+CoreTestsPigeonTestHostIntegrationCoreApiEchoNullableMapResponse*
+core_tests_pigeon_test_host_integration_core_api_echo_nullable_map_response_new(
+    FlValue* return_value);
+
+CoreTestsPigeonTestHostIntegrationCoreApiEchoNullableMapResponse*
+core_tests_pigeon_test_host_integration_core_api_echo_nullable_map_response_new_error(
+    const gchar* code, const gchar* message, FlValue* details);
+
+G_DECLARE_FINAL_TYPE(
+    CoreTestsPigeonTestHostIntegrationCoreApiEchoNullableEnumResponse,
+    core_tests_pigeon_test_host_integration_core_api_echo_nullable_enum_response,
+    CORE_TESTS_PIGEON_TEST,
+    HOST_INTEGRATION_CORE_API_ECHO_NULLABLE_ENUM_RESPONSE, GObject)
+
+CoreTestsPigeonTestHostIntegrationCoreApiEchoNullableEnumResponse*
+core_tests_pigeon_test_host_integration_core_api_echo_nullable_enum_response_new(
+    CoreTestsPigeonTestAnEnum* return_value);
+
+CoreTestsPigeonTestHostIntegrationCoreApiEchoNullableEnumResponse*
+core_tests_pigeon_test_host_integration_core_api_echo_nullable_enum_response_new_error(
+    const gchar* code, const gchar* message, FlValue* details);
+
+G_DECLARE_FINAL_TYPE(
+    CoreTestsPigeonTestHostIntegrationCoreApiEchoOptionalNullableIntResponse,
+    core_tests_pigeon_test_host_integration_core_api_echo_optional_nullable_int_response,
+    CORE_TESTS_PIGEON_TEST,
+    HOST_INTEGRATION_CORE_API_ECHO_OPTIONAL_NULLABLE_INT_RESPONSE, GObject)
+
+CoreTestsPigeonTestHostIntegrationCoreApiEchoOptionalNullableIntResponse*
+core_tests_pigeon_test_host_integration_core_api_echo_optional_nullable_int_response_new(
+    int64_t* return_value);
+
+CoreTestsPigeonTestHostIntegrationCoreApiEchoOptionalNullableIntResponse*
+core_tests_pigeon_test_host_integration_core_api_echo_optional_nullable_int_response_new_error(
+    const gchar* code, const gchar* message, FlValue* details);
+
+G_DECLARE_FINAL_TYPE(
+    CoreTestsPigeonTestHostIntegrationCoreApiEchoNamedNullableStringResponse,
+    core_tests_pigeon_test_host_integration_core_api_echo_named_nullable_string_response,
+    CORE_TESTS_PIGEON_TEST,
+    HOST_INTEGRATION_CORE_API_ECHO_NAMED_NULLABLE_STRING_RESPONSE, GObject)
+
+CoreTestsPigeonTestHostIntegrationCoreApiEchoNamedNullableStringResponse*
+core_tests_pigeon_test_host_integration_core_api_echo_named_nullable_string_response_new(
+    const gchar* return_value);
+
+CoreTestsPigeonTestHostIntegrationCoreApiEchoNamedNullableStringResponse*
+core_tests_pigeon_test_host_integration_core_api_echo_named_nullable_string_response_new_error(
+    const gchar* code, const gchar* message, FlValue* details);
+
+G_DECLARE_FINAL_TYPE(CoreTestsPigeonTestHostIntegrationCoreApi,
+                     core_tests_pigeon_test_host_integration_core_api,
+                     CORE_TESTS_PIGEON_TEST, HOST_INTEGRATION_CORE_API, GObject)
+
+typedef struct {
```

Comment:
We should probably have the generator generate a comment saying that this is the interface that must be implemented by the host API handler.